frontend/src/components/Schedule/Schedule.tsx

```diff
@@ -163,6 +168,35 @@ Rider: ${ride.rider.firstName} ${ride.rider.lastName}`,
 
   const closeModal = () => setIsOpen(false);
 
+  const cancelRide = (rideId : string) => {
+    fetch(`/api/rides/${rideId}`, withDefaults())
+    .then((res) => res.json())
+    .then((data) => {
+      if (data.recurring) {
+        fetch(
+          `api/rides/${rideId}/edits`,
+          withDefaults({
+            method: 'PUT',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ 'deleteOnly': 'true', 'origDate': scheduleDay}),
+          }),
+        )
+        .then(() => getRides())
+        .then(closeModal);
+      } else {
+        fetch(
+          `/api/rides/${rideId}`,
+          withDefaults({
+            method: 'DELETE',
+            headers: { 'Content-Type': 'application/json' },
+          }),
+        )
+        .then(() => getRides())
+        .then(closeModal);
+      }
+    })
+  }
```

Comment:
couldnt you pass the ride object instead of refetching the ride?

Reply:
I had to refetch the ride because type Ride doesn't have the recurring field. I could add it but I figured that would cause errors elsewhere.

Reply:
ah i see, if you can add it that would be good, but if theres errors you can cast it to `any` just as a temp fix for the time being